pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -106,7 +106,12 @@ in rec {
                                       ++ depsTargetTarget ++ depsTargetTargetPropagated) == 0;
       dontAddHostSuffix = attrs ? outputHash && !noNonNativeDeps || (stdenv.noCC or false);
       supportedHardeningFlags = [ "fortify" "stackprotector" "pie" "pic" "strictoverflow" "format" "relro" "bindnow" ];
-      defaultHardeningFlags = if stdenv.hostPlatform.isMusl
+                              # Musl-based platforms will keep "pie", other platforms will not.
+      defaultHardeningFlags = if stdenv.hostPlatform.isMusl &&
+                                # Except when:
+                                #    - static aarch64, where compilation works, but produces segfaulting dynamically linked binaries.
+                                #    - static armv7l, where compilation fails.
```

Comment:
Yes, the comment indent is weird.

It is not an error, but to keep the relevant comments close to the code.

If I didn't make the change to be the least intrusive in diffs, it would have looked like:

```nix
{/*...*/

defaultHardeningFlags =
  # Musl-based platforms will keep "pie", other platforms will not.
  if stdenv.hostPlatform.isMusl &&
    # Except when:
    #    - static aarch64, where compilation works, but produces segfaulting dynamically linked binaries.
    #    - static armv7l, where compilation fails.
      !((stdenv.hostPlatform.isAarch64 || stdenv.hostPlatform.isAarch32) && stdenv.hostPlatform.isStatic)
  then supportedHardeningFlags
  else lib.remove "pie" supportedHardeningFlags;

/*...*/}
```

I'm not exactly sure how I should format the comments around here to keep the relevant bits at the right location.

Reply:
I think it may be a good idea to place the condition inside an`( with stdenv.hostPlatform; /* ... */ )` block, e.g.:
```nix
{/*...*/
      defaultHardeningFlags = if (with stdenv.hostPlatform; isMusl &&
                                # Except when:
                                #    - static aarch64, where compilation works, but produces segfaulting dynamically linked binaries.
                                #    - static armv7l, where compilation fails.
                                !((isAarch64 || isAarch32) && isStatic))
                              then supportedHardeningFlags
                              else lib.remove "pie" supportedHardeningFlags;
/*...*/}
```